bash_utils/config.sh

```diff
@@ -165,6 +165,24 @@ function csv_export {
 }
 
 
+function csv_export_drop_columns {
+    local table=${1}
+    local columns=${2} #expected in format `csv_export_drop_columns mytable "'geom', 'other_column'"
+    local output_file=${3:-${table}}
+    local select_columns=$(run_sql_command \
+        "\COPY (SELECT '\"' || STRING_AGG(attname, '\",\"' ORDER BY attnum) || '\"'\
+        FROM pg_attribute\
+        WHERE attrelid = 'public.${table}'::regclass\
+            AND attname not in (${columns})\
+            AND attnum>0
+        ) TO STDOUT;")
+    run_sql_command \
+        "\COPY (\
+            SELECT ${select_columns} FROM ${table}\
+        ) TO STDOUT DELIMITER ',' CSV HEADER;">${output_file}.csv
```

Comment:
Don't love this entirely because in general it probably makes more sense (say in a dbt world) to simply make tables/views for all exports. But there's a specific case in which it's useful - tables where we export both a shapefile (with geom, shape_area, shape_leng. Or just geom), and a csv where we want to drop spatial fields. So in this case, it was nice to not make duplicate tables just without the geometries. Interested in your thoughts

Reply:
I feel you. I like this approach! especially using tables like `pg_attribute`. for future us, a quick comment above the `local select_columns= ...` line to explain what it's doing would be helpful